### PR TITLE
Add Azure tutorial that shows Elastic Agent

### PR DESCRIPTION
**Pull request by dedemorton**
Preview: https://observability-docs_2174.docs-preview.app.elstc.co/guide/en/observability/master/monitor-azure-elastic-agent.html

Open issues and questions:

- [x] **@zmoog** In the Elastic Agent tutorial, which integrations do we want to showcase? Right now we only show billing metrics. TBH, when I look at Kibana, I can't tell if there is a high level integration we can use to capture all the things. Need to know specifically which integrations we want to showcase in this tutorial. Should be at least one for metrics (billing?) and one for logs.
  _Decision: We will cover activity logs or Active Directory logs...not both. DeDe will make final decision._
- [x]  **@zmoog** What do users need to do to send logs to an event hub and which of these steps should we document in the tutorial?
  _Outcome: DeDe and Maurizio met to go through each scenario. DeDe will go through the steps and document them._
  - Do users need to follow all the steps here to create an event hub? https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-create
  - To collect Active Directory Logs, do users need to do all the steps here? https://learn.microsoft.com/en-us/azure/active-directory/reports-monitoring/tutorial-azure-monitor-stream-logs-to-event-hub
  - To collect activity logs, which of these steps do users need to do? (The info on this page is hard to follow) https://docs.microsoft.com/en-us/azure/azure-monitor/platform/activity-log-export 
- [x] Does everyone agree that putting the native Azure integration steps in a separate topic makes sense? _yes_
- [x] **@zmoog** Are the details in the [cloud docs](https://www.elastic.co/guide/en/cloud/current/ec-azure-marketplace-native.html) accurate? I would like to point to this content for more info about the native Azure integration.
  _Decision: We'll assume that these docs are correct and point to them._

Plus all the comments here: https://github.com/elastic/observability-docs/pull/2174#issuecomment-1263038216

TODO before merging:

- [x] Make sure the tutorial takes the user input from https://github.com/elastic/tech-content/issues/237 into consideration.
- [x] Look at the doc changes here to make sure we're consistent: https://github.com/elastic/integrations/pull/4219
- [x] Test the native integration steps.
- [x] Test the Elastic Agent steps in an isolated environment, if possible.

TODO after merging:

- [ ] Decide whether to group tutorials in the TOC by nesting them under a container. (Maybe defer until we get more of these tutorials updated.)
- [ ] Update Beats tutorial with Azure steps that have changed (maybe a follow-on issue + PR because that's outside the scope of this effort.

### CODE REVIEW

**Comment by zmoog** [comment]
> which integrations do we want to showcase?

We have four Azure integrations, each with one or more data streams:

- Azure Logs
  - Activity logs
  - Active Directory logs
    - Sign-in logs
    - Audit logs
  - Firewall logs
  - Platform logs
  - Spring Cloud logs (soon to be renamed Spring App logs)
  - Identity Protection logs (coming soon)
  - Provisioning logs (coming soon)
- Azure Metrics
  - Compute VM
  - Container Instance
  - Container Registry
  - Container Service
  - ...
- Azure Application Insights Metrics
  - Application Insights Metrics
  - Application State Insights Metrics
- Azure Billing Metrics
  - Billing Metrics

For the Azure Logs, from my experience, the most used ones are the Activity logs and the two Active Directory logs (sign-in and audit logs). I would pick one from Sign-in or Activity logs.

**Comment by dedemorton** [comment]
@zmoog From reading our Azure Logs integration docs, it looks like the steps for exporting activity logs to an event hub (using legacy collection) are quite different from exporting Active Directory logs. That makes me wonder if we should cover activity _and_ Active Directory logs. On the other hand, the steps for exporting Active Directory logs look more straightforward and recommending something called legacy collection seems like a bad idea. WDYT?

**Comment by zmoog** [comment]
> That makes me wonder if we should cover activity _and_ Active Directory logs. 

IMO we should cover activity _or_ Active Directory logs.

> On the other hand, the steps for exporting Active Directory logs look more straightforward and recommending something called legacy collection seems like a bad idea.

Activity and Directory logs are VERY similar from the integration perspective; the only difference is the source of the logs. 

Here's a quick diagram I am working on for the [Azure Logs revamp](https://github.com/elastic/integrations/pull/4300) I started today:

```text
  ┌────────────────┐   ┌──────────────┐   ┌────────────────┐                    
  │    Azure AD    │   │  Diagnostic  │   │ Azure AD logs  │                    
  │  <<service>>   │──▶│   settings   │──▶│ <<event hub>>  │──┐                 
  └────────────────┘   └──────────────┘   └────────────────┘  │   ┌────────────┐
                                                              │   │  Elastic   │
                                                              ├──▶│   Agent    │
  ┌────────────────┐   ┌──────────────┐   ┌────────────────┐  │   └────────────┘
  │ Azure Monitor  │   │  Diagnostic  │   │ Activity logs  │  │                 
  │  <<service>>   ├──▶│   settings   │──▶│ <<event hub>>  │──┘                 
  └────────────────┘   └──────────────┘   └────────────────┘                                
```

**Comment by zmoog** [comment]
About what users need to do to send logs to an event hub, we can check what I am writing for the already mentioned [Azure Logs doc revamp](https://github.com/elastic/integrations/pull/4300).

I believe there's an overlap between the two documents, but you can probably give me some advice about what goes in the tutorial and what goes in the general doc.

**Comment by zmoog** [comment]
> Does everyone agree that putting the native Azure integration steps in a separate topic makes sense?

Yeah! AFAIK the Native Azure Integration does not require an Agent, and we should handle it differently.

**Comment by zmoog** [comment]
> Are the details in the [cloud docs](https://www.elastic.co/guide/en/cloud/current/ec-azure-marketplace-native.html) accurate

Personally, I do not have a huge experience in the Native Azure Integration, but AFAIK they are accurate.

**Comment by dedemorton** [comment]
@zmoog  The content in this topic is ready to review: https://observability-docs_2174.docs-preview.app.elstc.co/guide/en/observability/master/monitor-azure-elastic-agent.html.

I think it's OK for the tutorial to focus on the big stuff and point to users for more detail in the integrations documentation. I'm not going to try to sync up the tasks we cover because the tutorial style is different and the tasks won't stay in sync anyhow. So try to review for accuracy, not total consistency with what you've written. :-) 

I have some remaining questions for you to consider as you review the content:

- [ ] Should we state that Elastic Agent 8.4.2 or later is a prereq? (8.4.2 is required to see forecast data because there is a bug in earlier versions.) We don't usually mention bug fix releases, tho.

- [ ] Under "Create an Azure service principal," is this still the recommended way to control access? I see a note on the Azure page for app registrations, but I don't know if it's relevant: "Starting June 30th, 2020 we will no longer add any new features to Azure Active Directory Authentication Library (ADAL) and Azure AD Graph. We will continue to provide technical support and security updates but we will no longer provide feature updates. Applications will need to be upgraded to Microsoft Authentication Library (MSAL) and Microsoft Graph." 

- [ ] Under "Grant access permission for your service principal," exactly which permissions are required? My setup had Reader and Billing Reader, but I don’t know if they are both required.

- [ ] Under "Step 5: Visualize Azure activity logs," the "failed to find message" lines in the screen grab are not a good look, but I don't know how to fix them. This was a problem for the AWS logs too. 

- [ ] Under "Configure diagnostic settings to send logs to the event hub," should we tell users which activity logs to collect? I am not sure myself.

- [ ] Under "Configure the Azure Logs integration to collect activity logs," do users specify the the connection string of the namespace OR the event hub?

**Comment by dedemorton** [comment]
@zmoog I am running into an error when I try to test the steps in the tutorial about using the native integration. When I try to create an Elasticsearch (Elastic Cloud) resource, I get the following message:
```
{"code":"DeploymentFailed","message":"At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.","details":[{"code":"500","message":"There is an existing Elastic Cloud account associated with your email address that has active deployments not running Azure. Please log in to [cloud.elastic.co](http://cloud.elastic.co/) and remove any deployments that are not running on Azure and try again. Contact [support@elastic.co](mailto:support@elastic.co) if you need further assistance."}]}
```

I *really* wanted to get this done today, so I deleted all my deployments on cloud, but I am still getting the same message. I even tried creating a deployment in Elastic Cloud on Azure to "grease the skids" but no luck. I feel like I saw something somewhere about issues related to our elastic cloud accounts, but can't remember the details. Can you help?

I'm blocked on testing the native steps until I figure out what's wrong.

File: docs/en/observability/monitor-azure-native.asciidoc [hunk 1]
```diff
@@ -0,0 +1,155 @@
+[[monitor-azure-native]]
+== Monitor Microsoft Azure with the native Azure integration
```
Comment (dedemorton) [inline docs/en/observability/monitor-azure-native.asciidoc:2]: This file is not yet ready for review.

**Comment by zmoog** [comment]
> Should we state that Elastic Agent 8.4.2 or later is a prereq? (8.4.2 is required to see forecast data because there is a bug in earlier versions.) We don't usually mention bug fix releases, tho.

Forecast data stopped working reliably right after we shipped the upgrade. There is a backport in progress to 7.17.7, but version 8.4.2 is required to have it working across the existing Azure account types. 

What about recommending using the latest Agent version for 7.x and 8.x to get support for the latest API changes from Azure?

**Comment by zmoog** [comment]
> Under "Create an Azure service principal," is this still the recommended way to control access?

Not 100% sure here. I believe app registration is still the way to go, but I will double-check if the upcoming changes will affect app registration or is an authentication library detail only.

I didn't know this post, but we were aware of the in-progress transition to the Microsoft Identity service due to changes in the Azure SDK for Go we use to access Azure services from integration and Beats. Thank you for bringing this up! The post adds more details to the context.

The new Azure SDK for Go library that supports Microsoft Identity [^1] requires Go version 1.18. Fortunately Go 1.18 landed in 7.x and 8.x and we can finally move forward with the plans to switch to `azidentity`.

[^1]: The library is called `azidentity`.

**Comment by zmoog** [comment]
> Under "Grant access permission for your service principal," exactly which permissions are required? My setup had Reader and Billing Reader, but I don't know if they are both required.

I understand that Azure Billing Metrics only requires the built-in "Billing Reader" role assignment. The "Reader" role is a broader role that gives the app access to a much wider set of information.

The "Reader" role gives the app access to 5991 individual permissions across multiple Azure services. The "Billing Reader" rolprovides the app withthe app access to 86 permissions.

You can check the permissions list by:

* Visit our Azure subscription > Access Control (IAM)
* Select the "Role" tab
* Filter the role name in the "search by" field
* Click on the "View" link in the "Details" column

Azure Billing Metrics only requires a handful of permissions. I think we could write instructions about how to write a custom role with the strictly required permissions only. I believe using the Azure built-in role is okay from a security standpoint and can provide a better user experience. 

However, we could also provide a list with all the permissions required, so security-inclined customers have the option to nail down the exact permission set the app needs to work.

@dedemorton WDYT?

**Comment by zmoog** [comment]
> Under "Step 5: Visualize Azure activity logs," the "failed to find message" lines in the screen grab are not a good look, but I don't know how to fix them. This was a problem for the AWS logs too.

Yeah, it does not look good. This probably happens because the Logs stream page looks for a field named "message" by default, but Azure logs (and probably also AWS logs) don't have such a field.

I think this can be customized in the Logs stream option, so we and the users can pick an existing and meaningful field (for example, "category" or something specific to this log category).

**Comment by zmoog** [comment]
>  Under "Configure diagnostic settings to send logs to the event hub," should we tell users which activity logs to collect? I am not sure myself.

Yeah, we added tables with a detailed list of supported log categories in the recent Azure Logs documentation updates. Maybe we can point users to those tables to get the most up-to-date version while we update existing integration and add new ones? Not sure we have Activity logs yet but are working on it so we can add them.

**Comment by zmoog** [comment]
> Under "Configure the Azure Logs integration to collect activity logs," do users specify the connection string of the namespace OR the event hub?

The connection string is on the event hub namespace:

Event hub namespace > NAMESPACE > Shared access policies > POLICY > key > Connection string

Since we met last, I expanded the even hub setup portion in https://github.com/elastic/integrations/pull/4300 to better describe this step. Let me know if you think the doc describes this step well enough.

**Comment by dedemorton** [comment]
@zmoog  OK, I think I've figured out how to resolve most of my open issues. I have a couple follow-up questions:

* Regarding roles/permissions, it's probably best to recommend a built-in role (to avoid making the tutorial even longer). I think you are saying that  the `Billing Reader` role is required, but not `Reader`. Is that correct?  I don't have time to run through the steps all over again. :-)

* For the diagnostic settings, can you just confirm that the categories I show in the [screen capture here](https://observability-docs_2174.docs-preview.app.elstc.co/guide/en/observability/master/monitor-azure-elastic-agent.html#azure-elastic-agent-configure-azure-diagnostics) are supported? I can add a link to the Azure logs integration docs later.

File: docs/en/observability/monitor-azure-agent.asciidoc [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+[[azure-collect-metrics]]
+=== Step 1: Create an Azure service principal
+
+In this step, you create an Azure service principal and then grant them access
+to use the Azure Monitor REST API.
+
+The https://docs.microsoft.com/en-us/rest/api/monitor/[Azure Monitor REST API]
+allows you to get insights into your Azure resources using different operations.
+To access the Azure Monitor REST API, you need to use the Azure Resource Manager
+authentication model. Therefore, you must authenticate all requests with Azure
```
Comment (zmoog) [inline docs/en/observability/monitor-azure-agent.asciidoc:37]: @dedemorton, in this section, you mention "Azure Monitor REST API." 

I believe we can use the more general name "Azure REST API." The [Azure REST API](https://learn.microsoft.com/en-us/rest/api/azure/) is not a specific API endpoint, but the collection of all actual endpoints and [Azure Monitor REST API](https://learn.microsoft.com/en-us/rest/api/monitor/) is one of them.  

For example, the Azure Billing Metrics uses two different Azure APIs: Consumption API and Cost Management API.

Depending on their permission, app registrations can access a wide range of APIs.

File: docs/en/observability/monitor-azure-agent.asciidoc [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+**Tenant ID**:: The tenant ID listed on the main Azure Active Directory Page.
+
+**Subscription ID**:: The subscription ID listed on the main Subscriptions page.
+--
++
+[role="screenshot"]
+image::images/agent-tut-azure-integration-settings.png[Screenshot of integration settings for Azure]
+
+. Make sure the **Collect Azure Billing metrics** selector is turned on, and
+optionally change the default period to a multiple of 24h.
```
Comment (zmoog) [inline docs/en/observability/monitor-azure-agent.asciidoc:153]: The 24h period should be threaded like an advanced setting. The integration makes some assumptions about the period, so users should not change this value unless they know what they are doing.

**Comment by zmoog** [comment]
> * Regarding roles/permissions, it's probably best to recommend a built-in role (to avoid making the tutorial even longer). I think you are saying that the `Billing Reader` role is required, but not `Reader`. Is that correct?  I don't have time to run through the steps all over again. :-)

I used `Billing Reader` during all my tests, so I'm confident it is a good fit for the Azure Billing Metrics integration.

> For the diagnostic settings, can you just confirm that the categories I show in the [screen capture here](https://observability-docs_2174.docs-preview.app.elstc.co/guide/en/observability/master/monitor-azure-elastic-agent.html#azure-elastic-agent-configure-azure-diagnostics) are supported? I can add a link to the Azure logs integration docs later.

I don't have a list ready now, but I think the log categories listed are supported. Once complete, the list with the supported log categories will show up in the internal integration doc in Kibana and https://docs.elastic.co/integrations/azure/activitylogs.

File: docs/en/observability/monitor-azure-agent.asciidoc (hunk 3 above)
Comment (dedemorton) [reply docs/en/observability/monitor-azure-agent.asciidoc:153]: Sounds good. I won't mention this setting in the tutorial.

**Comment by dedemorton** [comment]
@zmoog And now for the latest chapter in the saga "All the Azure Monitoring stuff that doesn't work as expected": I am not getting activity logs when I use the native Azure integration. 

image


The diagnostics settings in azure look OK (I think):
image

I followed the steps in the tutorial and deployed the resource before enabling Logs & Metrics. Do I need to redeploy after enabling those options? (I am seeing platform logs, so I don't think this is my problem.)

**Comment by dedemorton** [comment]
@zmoog Native steps are updated! I'm not going to worry about the activity logs not showing up right now, but someone needs to investigate to see if there's a software issue or the steps are wrong.

This tutorial is ready for a final review. Thanks again for your help with this.

**Comment by zmoog** [comment]
> I'm not going to worry about the activity logs not showing up right now, but someone needs to investigate to see if there's a software issue or if the steps are wrong.

Hey @dedemorton, I check the activity logs and create an issue if something is wrong or get back to you if there's something related to the tutorial.

File: docs/en/observability/monitor-azure-agent.asciidoc [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+=== Step 5: Visualize Azure activity logs
+
+Now that logs are streaming into {es}, you can visualize them in {kib}. To see
+the raw logs, open the main menu in {kib}, then click **Logs**. Notice that you
+can filter on a specific data stream. This example uses
+`data_stream.dataset : "azure.activitylogs"` to show Azure activity logs:
+
+[role="screenshot"]
+image::images/agent-tut-azure-activity-logs.png[Screenshot of Logs app showing Azure activity logs]
+
```
Comment (zmoog) [inline docs/en/observability/monitor-azure-agent.asciidoc:371]: Nit: we can avoid the "failed to find message" by suggesting users change the Observability > Logs > Stream settings:

CleanShot 2022-10-10 at 17 43 02@2x


CleanShot 2022-10-10 at 17 43 17@2x

File: docs/en/observability/monitor-azure-native.asciidoc [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+[discrete]
+[[azure-create-resource]]
+=== Step 1: Create an {es} resource in the Azure portal
+
+IMPORTANT: These steps will not work if you have an active GCP or AWS deployment
+in {ecloud} that is already associated with the email address used for your
+Azure account. To avoid this problem, delete your GCP and AWS deployments in
+{ecloud}, or use a different Azure account. If this does not resolve your issue,
+reach out to us at support@elastic.co.
```
[the remark this answers is not in the document]
Comment (dedemorton) [reply docs/en/observability/monitor-azure-native.asciidoc:31]: If we suggest a time, we should be specific like...wait nn hours. Is 24 hours enough? 12 hours?

File: docs/en/observability/monitor-azure-agent.asciidoc (hunk 4 above)
Comment (dedemorton) [reply docs/en/observability/monitor-azure-agent.asciidoc:371]: I'll add that as a TIP.  I'll leave the screen capture because I think it's actually useful to explain why this happens.

File: docs/en/observability/monitor-azure-native.asciidoc (hunk 5 above)
[the remark this answers is not in the document]
Comment (zmoog) [reply docs/en/observability/monitor-azure-native.asciidoc:31]: I'm leaning towards 12 hours, but @csierra may have a better take on this.

File: docs/en/observability/monitor-azure-agent.asciidoc (hunk 4 above)
Comment (zmoog) [reply docs/en/observability/monitor-azure-agent.asciidoc:371]: Yeah, it's the right call since it's what most users will get OOTB.

File: docs/en/observability/monitor-azure-native.asciidoc (hunk 5 above)
[the remark this answers is not in the document]
Comment (csierra) [reply docs/en/observability/monitor-azure-native.asciidoc:31]: hi folks... I am not sure I am the most knowledgeable person about this... I don't really know how long could it take since a deployment is requested to shutdown until it actually does shutdown completely for the rest of the systems

[the remark this answers is not in the document]
Comment (dedemorton) [reply docs/en/observability/monitor-azure-native.asciidoc:31]: @csierra Is there some way for users to make sure the deployment is shutdown completely? Otherwise I think we'll need to be really vague in the docs (not a great user experience).

[the remark this answers is not in the document]
Comment (csierra) [reply docs/en/observability/monitor-azure-native.asciidoc:31]: @dedemorton I think I found a bug in the logic accessing the existing deployments that may explain the delay that you have experienced. 
Let me make a couple of inquiries and I will come back to you folks on this.

[the remark this answers is not in the document]
Comment (csierra) [reply docs/en/observability/monitor-azure-native.asciidoc:31]: so I am not sure if it's a bug but I believe the problem is that we seem to consider terminated deployments that are NOT hidden as incompatible. So even if you stopped your deployment, if it was not hidden, it was probably still considered as incompatible and yielding an error.

[the remark this answers is not in the document]
Comment (csierra) [reply docs/en/observability/monitor-azure-native.asciidoc:31]: I have tried this situation myself and it seems it works instantaneously if the user deletes their deployment from the cloud console. Once the deployment reads that it is hidden it is no longer taken into account by the marketplace validation and it just passes through. 
I have successfully tried this in staging.

[the remark this answers is not in the document]
Comment (dedemorton) [reply docs/en/observability/monitor-azure-native.asciidoc:31]: Thanks for following up on this! In my case, I simply deleted the deployments, but it sounds like there might have been an unexpected glitch. I will leave the tutorial as-is for now and update it later if we get feedback indicating that users are running into the same problem I encountered.
